pkgs/by-name/ni/nixos-firewall-tool/nixos-firewall-tool-nftables.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+show_help() {
+    echo "nixos-firewall-tool"
+    echo ""
+    echo "Can temporarily manipulate the NixOS firewall"
+    echo ""
+    echo "Open TCP port:"
+    echo " nixos-firewall-tool open tcp 8888"
+    echo ""
+    echo "Show all firewall rules:"
+    echo " nixos-firewall-tool show"
+    echo ""
+    echo "Open UDP port:"
+    echo " nixos-firewall-tool open udp 51820"
+    echo ""
+    echo "Reset firewall configuration to system settings:"
+    echo " nixos-firewall-tool reset"
+}
+
+if [[ -z ${1+x} ]]; then
+  show_help
+  exit 1
+fi
+
+case $1 in
+  "open")
+    protocol="$2"
+    port="$3"
+
+    nft add element inet nixos-fw "$protocol-ports" "{ $port }"
+  ;;
+  "show")
+    nft list table inet nixos-fw
+  ;;
+  "reset")
+    systemctl reload nftables.service
```

Comment:
Reloading nftables for undoing port changes is a bit excessive, see: https://github.com/NixOS/nixpkgs/issues/286584.

Instead of the current approach, I would suggest add a set of type `inet_proto . inet_service`, call it something like `tool-ports`, and let this command do a single flush of this set.